grin/src/adapters.rs

```diff
@@ -279,37 +266,23 @@ impl NetToChainAdapter {
 			peer_store: peer_store,
 			connected_peers: connected_peers,
 			tx_pool: tx_pool,
-			syncer: OneTime::new(),
 		}
 	}
 
-	/// Start syncing the chain by instantiating and running the Syncer in the
-	/// background (a new thread is created).
-	pub fn start_sync(&self, sync: sync::Syncer) {
-		let arc_sync = Arc::new(sync);
-		self.syncer.init(arc_sync.clone());
-		let _ = thread::Builder::new()
-			.name("syncer".to_string())
-			.spawn(move || {
-				let res = arc_sync.run();
-				if let Err(e) = res {
-					panic!("Error during sync, aborting: {:?}", e);
-				}
-			});
-	}
-
-	pub fn syncing(&self) -> bool {
-		self.syncer.is_initialized() && self.syncer.borrow().syncing()
-	}
-
 	/// Prepare options for the chain pipeline
+	/// SYNC: do not broadcast block out to all our peers (we want to sync quietly)
+	/// NONE: regular block so broadcast it out and keep the network up to date
 	fn chain_opts(&self) -> chain::Options {
-		let opts = if self.syncing() {
-			chain::SYNC
-		} else {
-			chain::NONE
-		};
-		opts
+		let header_head = self.chain.get_header_head();
+		let block_header = self.chain.head_header();
+		if let Ok(header) = header_head {
+			if let Ok(block_header) = block_header {
+				if header.height < block_header.height + 100 {
```

Comment:
Looks like this is the wrong comparison direction.

Reply:
Thanks! Good catch.